sw/airborne/subsystems/ins/ins_int.c

```diff
@@ -553,9 +583,25 @@ static void vel_est_cb(uint8_t sender_id __attribute__((unused)),
 #else
   ins_int.ltp_speed.x = SPEED_BFP_OF_REAL(vel_ned.x);
   ins_int.ltp_speed.y = SPEED_BFP_OF_REAL(vel_ned.y);
+
+  // due to time step resolution, we average velocity over longer time window for int position resolution
   if (last_stamp > 0) {
-    ins_int.ltp_pos.x = ins_int.ltp_pos.x + POS_BFP_OF_REAL(dt * vel_ned.x);
-    ins_int.ltp_pos.y = ins_int.ltp_pos.y + POS_BFP_OF_REAL(dt * vel_ned.y);
+    // integrate area under speed curve assuming constant gradient between speed measurements (ie trapezoid)
+    sum_x += (prev_vel.x + vel_ned.x) * (stamp - last_stamp) / 2.;
+    sum_y += (prev_vel.y + vel_ned.y) * (stamp - last_stamp) / 2.;
+  } else {
+    last_pos_stamp = stamp;
+  }
+
+  last_stamp = stamp;
+  prev_vel.x = vel_ned.x;
+  prev_vel.y = vel_ned.y;
+
+  if (stamp - last_pos_stamp >= 0.05) {
+    ins_int.ltp_pos.x += POS_BFP_OF_REAL(sum_x * 1e-6);
+    ins_int.ltp_pos.y += POS_BFP_OF_REAL(sum_y * 1e-6);
+    sum_x = 0.; sum_y = 0.;
+    last_pos_stamp = stamp;
   }
```

Comment:
I think I will put this in a separate PR

Reply:
Thx for reviving